Add CPS ML datafeed troubleshooting docs for serverless (#7403)
elastic/docs-content: troubleshoot/elasticsearch/machine-learning/cps-datafeed-project-scope.md

Reviewer: This page covers four distinct problems but groups them under two mega-sections (Diagnose, Resolve). Readers have to mentally map which fix applies to which scenario. I suggest that you restructure so each problem is self-contained (what you see, fix, verify):

- Routing matches no project 
- Scope is wider than intended
- Stale project reference
- Cloned job inherits wrong scope

Consider moving the shared preconditions (stop datafeed, rollback gate) into a "Before you update" section at the top because they apply to all four problems.

Author: Done in 2dc81b8b3 — page split into four problem sections with shared Before you update at top.

> §§§§§§§§§§§§§§§§§§§§§§§---

Reviewer: ```suggestion
---
```

Author: Fixed in 2dc81b8b3 — frontmatter now starts with `---`.

navigation_title: Project scope problems
applies_to:
  stack: unavailable
  serverless: preview
products:
  - id: elasticsearch
  - id: cloud-serverless
  - id: machine-learning
---

> # Project scope problems [cps-datafeed-project-scope]

Reviewer: this title needs to be more specific

no need for anchors on titles (will fix the helpers to stop doing that) 

```suggestion
# Troubleshoot anomaly detection datafeed project scope
```

Author: Done in 2dc81b8b3 — H1 is now Troubleshoot anomaly detection datafeed project scope (no anchor).

The `project_routing` field decides which linked projects a {{cps}} {{dfeed}} searches. This page covers routing that matches nothing, routing that matches too many projects, references to projects that no longer exist under an alias, and clones that inherit the source {{anomaly-job}}'s scope.

## Diagnose project scope problems [diagnose-cps-datafeed-project-scope]

:::{include} /troubleshoot/_snippets/cps-ml-diagnostics-sources.md
:::

> 
> **Check the effective scope in {{kib}}**

Reviewer: prefer proper headings over bold

```suggestion

### Check the effective scope in {{kib}}
```

Author: Done in 2dc81b8b3 — bold section labels converted to `###` headings throughout the leaf pages.

> Open **Machine Learning → Anomaly Detection** and review the **Project scope** column in the {{anomaly-jobs}} list. Each cell shows a parsed/total count (for example, `2/5`) derived from the stored routing expression and the total project count (origin plus linked projects), not from live resolution of which aliases match. {{kib}} parses the segment after `_alias:`: omitted or `null` routing shows `1`. `_alias:*` shows the origin-plus-linked total. A single expression such as `_alias:production-*` or `_alias:_origin` shows `1`. Because the count comes from parsing rather than resolution, a wildcard that matches several projects still shows `1`. Select the count to open a popover titled **Project scope** that displays the routing expression stored on the {{dfeed}}. A legacy {{anomaly-job}} with no stored routing shows `_alias:_origin` in the popover.

Reviewer: can we optimize this to focus on the gotcha? wonder if this info should move to the conceptual docs because this is documenting a field in the UI

```suggestion
Open **Machine Learning > Anomaly Detection** and review the **Project scope** column. Each cell shows a parsed count out of the total project count (origin plus linked projects). For example, `2/5` means the expression targets 2 projects out of 5 available.

The parsed count comes from the routing expression text, not from resolving which aliases actually match at runtime:
- Omitted or `null` routing shows `1` (origin only).
- `_alias:*` shows the full origin-plus-linked total.
- `_alias:_origin` shows `1`.
- A single wildcard expression like `_alias:production-*` shows `1`, even if it matches several linked projects at runtime.

Select the count to open a popover that displays the routing expression stored on the datafeed. A legacy job with no stored routing shows `_alias:_origin` in the popover.
```

Author: Done in 2dc81b8b3 — condensed to the gotcha-focused Project scope column explanation.

| Qualified `project:index` references a project that does not exist or is unauthorized | Fails immediately on create or update |

Validate-before-mint defers a no-match `project_routing` to run time only when every entry in `indices` is unqualified (flat-world). Any qualified `project:index` pattern, including one whose project alias is missing, fails immediately on create or update.

### Fix

Update routing to a valid `_alias:` expression:

```console
POST _ml/datafeeds/{datafeed_id}/_update
{
  "project_routing": "_alias:production-*"
}
```

If routing references a project that should be in scope, establish the link in [Link and manage projects](/deploy-manage/cross-project-search-config/cps-config-link-and-manage.md), then update routing or wait for the next extraction cycle.

### Verify

* The **Project scope** column shows the intended parsed/total count and popover expression.
* `GET _ml/datafeeds/{datafeed_id}` returns the expected `project_routing` value.
* `GET _ml/datafeeds/{datafeed_id}/_stats` shows successful extraction cycles.

> ## Scope is wider than intended

Reviewer 2: Should we just have a more common "scope" documentation for all of CPS, and datafeeds can link to that? Otherwise, I imagine there would be a few pages with very similar information
